test/Swashbuckle.AspNetCore.SwaggerGen.Test/SwaggerGenJsonOptionsTests.cs

```diff
@@ -0,0 +1,105 @@
+    }
+
+    [Fact]
+    public static void Ensure_SwaggerGenJsonOptions_Uses_Mvc_JsonOptions_When_Overridden()
+    {
+        var expectedDummyConverter = new DummyConverter();
+
+        var services = new ServiceCollection();
+        services.AddSingleton<IWebHostEnvironment, DummyHostEnvironment>();
+        services.AddSwaggerGen();
+        services.ConfigureHttpJsonOptions(o => o.SerializerOptions.Converters.Add(new DummyConverter()));
+        services.AddMvcCore().AddJsonOptions(o => o.JsonSerializerOptions.Converters.Add(expectedDummyConverter));
+        services.AddSwaggerGenMvcJsonOptions();
+
+        using var provider = services.BuildServiceProvider();
+
+        var swaggerGenDummyConverter = provider.GetService<IOptions<SwaggerGenJsonOptions>>().Value.SerializerOptions.Converters.FirstOrDefault();
+
+        Assert.Equal(expectedDummyConverter, swaggerGenDummyConverter);
+    }
+
+    [Fact]
+    public static void Ensure_SwaggerGenJsonOptions_Uses_Mvc_JsonOptions_When_Not_Using_Minimal_Apis()
+    {
+        var expectedDummyConverter = new DummyConverter();
+
+        var services = new ServiceCollection();
+        services.AddSingleton<IWebHostEnvironment, DummyHostEnvironment>();
+        services.AddSwaggerGen();
+        services.AddMvcCore().AddJsonOptions(o => o.JsonSerializerOptions.Converters.Add(expectedDummyConverter));
+
+        using var provider = services.BuildServiceProvider();
+
+        var swaggerGenDummyConverter = provider.GetService<IOptions<SwaggerGenJsonOptions>>().Value.SerializerOptions.Converters.FirstOrDefault();
+
+        Assert.Equal(expectedDummyConverter, swaggerGenDummyConverter);
+    }
+
+    [Fact]
+    public static void Ensure_SwaggerGenJsonOptions_Uses_MinimalApi_JsonOptions_When_Configured()
+    {
+        var expectedDummyConverter = new DummyConverter();
+
+        var services = new ServiceCollection();
+        services.AddSingleton<IWebHostEnvironment, DummyHostEnvironment>();
+        services.AddSwaggerGen();
+        services.ConfigureHttpJsonOptions(o => o.SerializerOptions.Converters.Add(expectedDummyConverter));
+        services.AddMvcCore().AddJsonOptions(o => o.JsonSerializerOptions.Converters.Add(new DummyConverter()));
+
+        using var provider = services.BuildServiceProvider();
+
+        var swaggerGenDummyConverter = provider.GetService<IOptions<SwaggerGenJsonOptions>>().Value.SerializerOptions.Converters.FirstOrDefault();
+
+        Assert.Equal(expectedDummyConverter, swaggerGenDummyConverter);
+    }
+
+    private sealed class DummyHostEnvironment : IWebHostEnvironment
+    {
+        public string WebRootPath { get; set; }
+        public IFileProvider WebRootFileProvider { get; set; }
+        public string ApplicationName { get; set; }
+        public IFileProvider ContentRootFileProvider { get; set; }
+        public string ContentRootPath { get; set; }
+        public string EnvironmentName { get; set; }
+    }
```

Comment:
I think we have a few copies of this in the tests. If any are in the same project as this one, could you extract out into shared type that can be re-used?